controllers/operatorconfig_controller.go

```diff
@@ -84,21 +132,42 @@ func (r *OperatorConfigReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 	return ctrl.Result{}, err
 }
 
-func configureProxy(ctx context.Context, log logr.Logger) error {
-	return createResourcesFromFile("proxy.yaml", ctx, log)
+func (r *OperatorConfigReconciler)configureProxy(ctx context.Context, log logr.Logger, operatorConfig *cranev1alpha1.OperatorConfig) error {
+	return r.createResourcesFromFile("proxy.yaml", ctx, log, operatorConfig)
 }
 
-func configureCranePlugin(ctx context.Context, log logr.Logger) error {
-	return createResourcesFromFile("crane-ui-plugin.yaml", ctx, log)
+func (r *OperatorConfigReconciler)configureCranePlugin(ctx context.Context, log logr.Logger, operatorConfig *cranev1alpha1.OperatorConfig) error {
+	return r.createResourcesFromFile("crane-ui-plugin.yaml", ctx, log, operatorConfig)
 }
 
-func createClusterTasks(ctx context.Context, log logr.Logger) error {
-	return createResourcesFromFile("manifests.yaml", ctx, log)
+func (r *OperatorConfigReconciler)createClusterTasks(ctx context.Context, log logr.Logger, operatorConfig *cranev1alpha1.OperatorConfig) error {
+	return r.createResourcesFromFile("manifests.yaml", ctx, log, operatorConfig)
+}
+
+func (r *OperatorConfigReconciler) CleanUpOpjects(ctx context.Context) error {
+	err := r.deleteResourcesFromFile("proxy.yaml", ctx)
+	if err != nil {
+		return err
+	}
+
+	err = r.deleteResourcesFromFile("crane-ui-plugin.yaml", ctx)
+	if err != nil {
+		return err
+	}
+	err = r.deleteResourcesFromFile("manifests.yaml", ctx)
+	if err != nil {
+		return err
+	}
+	return nil
 }
 
 // SetupWithManager sets up the controller with the Manager.
 func (r *OperatorConfigReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&cranev1alpha1.OperatorConfig{}).
+		Owns(&v1.Deployment{}).
```

Comment:
is this the complete list of resources to watch?

Reply:
we should also manage Clustertask and consoleplugin, these two are clusterscoped, figuring out how to do that

Reply:
Added all the resources we need to be managing

Reply:
https://github.com/kubernetes-sigs/controller-runtime/issues/1296#issuecomment-752390912 PTAL I do not think Owns is the right function to be used when working with cluster scoped objects, since owner references don't work at cluster scope.

Reply:
https://kubernetes.io/docs/concepts/overview/working-with-objects/owners-dependents/#owner-references-in-object-specifications more details here, look at the `Note` section

Reply:
```
// SetupWithManager sets up the controller with the Manager.
func (r *OperatorConfigReconciler) SetupWithManager(mgr ctrl.Manager) error {
	pluginPredicate, err := predicate.LabelSelectorPredicate(v12.LabelSelector{
		MatchLabels: map[string]string{"label-key": "label-value"},
	}))
	if err != nil {
		return err
	}
	return ctrl.NewControllerManagedBy(mgr).
		For(&cranev1alpha1.OperatorConfig{}).
		Owns(&v1.Deployment{}).
		Owns(&corev1.Service{}).
		Owns(&corev1.ConfigMap{}).
		Watches(&source.Kind{Type: &v1alpha1.ConsolePlugin{}}, &handler.EnqueueRequestForObject{}, builder.WithPredicates(pluginPredicate)).
		Owns(&v1beta1.ClusterTask{}).
		Complete(r)
}
```

May be something like the above might help if you create the ConsolePlugin object with right labels

Reply:
Thanks :smiley:

Reply:
> https://github.com/kubernetes-sigs/controller-runtime/issues/1296#issuecomment-752390912 PTAL I do not think Owns is the right function to be used when working with cluster scoped objects, since owner references don't work at cluster scope.

@JaydipGabani pointed out to me that OperatorConfig is a clusterscoped resource, which means Owns should work, but the kube docs seem to be confusing, long-term would rather have a good e2e test to make sure we are covered here.